### PR TITLE
Add definition for RasterSymbolizer

### PR DESCRIPTION
**Pull request by jansule**
Adds an initial definition for a RasterSymbolizer based on SLD specification.

More specifically, defines:
- Opacity
- ColorMap
- ChannelSelection
- ContrastEnhancement

### CODE REVIEW

File: sample.ts [hunk 1]
```diff
@@ -91,6 +91,24 @@ const sampleStyle: Style = {
...
+            }
+          ],
+          extended: false,
+          type: "ramp"
+        },
+        channelSelection: {
+          grayChannel: {
+            sourceChannelName: '1'
+          }
+        }
```
Comment (KaiVolland) [inline sample.ts]: Might be better to create a seperate sample file for raster styles as mixing doesn't make sense, or does it?

Comment (jansule) [reply sample.ts]: According to the SLD Reference mixing is fine. Also checked it on geoserver and had no validation errors.

Comment (KaiVolland) [reply sample.ts]: I don't know any datasource that stores raster and vector data at the same time.
I think we also need an own `StyleType` for raster data `raster`, as mixing really makes no sense to me.
BTW: Do we use this property anywhere? :)

Comment (jansule) [reply sample.ts]: > as mixing really makes no sense to me.

I agree with you on that. I can create another example, that's not a problem. Anyway, we should also support mixing the types as it is not our concern if that makes sense for a user or not.

> BTW: Do we use this property anywhere? :)

We use it at least in the ol-parser. Will add the according property.

File: index.d.ts [hunk 2]
```diff
@@ -231,10 +231,79 @@ export interface LineSymbolizer extends BaseSymbolizer {
...
+  color: string;
+  quantity?: number;
+  label?: string;
+  opacity?: number;
+}
+
+/**
+ * The Types that are allowed in a ColorMap.
+ */
+export type ColorMapType = 'ramp'|'intervals'|'values';
```
Comment (dnlkoch) [inline index.d.ts]: Minor: Missing spaces.

File: index.d.ts [hunk 3]
```diff
@@ -231,10 +231,79 @@ export interface LineSymbolizer extends BaseSymbolizer {
...
 
+/**
+ * A single entry for the ColorMap.
+ */
+export interface ColorMapEntry {
+  color: string;
+  quantity?: number;
+  label?: string;
+  opacity?: number;
+}
```
Comment (dnlkoch) [inline index.d.ts]: All color map entries (at least for SLD, see [here](https://docs.geoserver.org/stable/en/user/styling/sld/reference/rastersymbolizer.html#cql-expressions)) allow for CQL expression. If we want to support this, we should support number and string inputs for all entries.

File: rastersample.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { Style } from 'index';
+
+const sampleRasterStyle: Style = {
+  name: 'Sample Raster Style',
+  rules: [
+    {
+      name: 'Very old Peter',
```
Comment (chrismayer) [inline rastersample.ts]: Are you really VERY OLD if you are beyond the age of 51? :wink:

File: index.d.ts (hunk 2 above)
Comment (jansule) [reply index.d.ts]: solved in https://github.com/terrestris/geostyler-style/pull/113/commits/c6a661d54274315ef2f4c4c5f1fe23898b2f6d1f

File: index.d.ts (hunk 3 above)
Comment (jansule) [reply index.d.ts]: Solved in https://github.com/terrestris/geostyler-style/pull/113/commits/62db216711613220b62772ad871f0e54b8728c9b

Comment (jansule) [reply index.d.ts]: Update: In coordination with @KaiVolland we decided to not support CQl expressions for now as they are geoserver specific and not part of the official standard. Reverted corresponding commit.

File: rastersample.ts (hunk 4 above)
Comment (marcjansen) [reply rastersample.ts]: Whats my age again?

**Pull request merged**